x-pack/plugins/secops/server/lib/network_top_n_flow/elasticsearch_adapter.ts

```diff
@@ -77,3 +50,44 @@ export class ElasticsearchNetworkTopNFlowAdapter implements NetworkTopNFlowAdapt
     };
   }
 }
+
+const getTopNFlowEdges = (
+  response: DatabaseSearchResponse<NetworkTopNFlowData, TermAggregation>,
+  options: NetworkTopNFlowRequestOptions
+): NetworkTopNFlowEdges[] => {
+  if (options.networkTopNFlowDirection === NetworkTopNFlowDirection.uniDirectional) {
+    return formatTopNFlowEgdes(
+      getOr([], 'aggregations.top_uni_flow.buckets', response),
+      options.networkTopNFlowType
+    );
+  }
+  return formatTopNFlowEgdes(
+    getOr([], 'aggregations.top_bi_flow.buckets', response),
+    options.networkTopNFlowType
+  );
+};
+
+const formatTopNFlowEgdes = (
+  buckets: NetworkTopNFlowBuckets[],
+  networkTopNFlowType: NetworkTopNFlowType
+): NetworkTopNFlowEdges[] =>
+  buckets.map((bucket: NetworkTopNFlowBuckets) => ({
+    node: {
+      _id: bucket.key,
+      timestamp: bucket.timestamp.value_as_string,
+      [networkTopNFlowType]: {
+        count: bucket.ip_count.value || null,
+        domain: bucket.domain.buckets.map(bucketDomain => bucketDomain.key) || null,
+        ip: bucket.key,
+      },
+      network: {
+        bytes: bucket.bytes.value || null,
+        packets: bucket.packets.value || null,
+        direction: bucket.direction.buckets.map(bucketDir => bucketDir.key) || null,
```

Comment:
All of these null setters here are probably not doing what you think they are. There's a lot of times where `ip_count.value`, `bytes.value`, and `packets.value` are going to be `0`.

When this is run on 0 values it will set the values to `null` instead of `zero` which will _could_ make the UI think that it getting back unknown values rather than `0` values.

If this is as intended and you do know that 'yes, for a fact' you want `0` transformed into null, I would advise against using `null` as a value for `0` since on the UI we have written utilities to designate `--` as empty non-existent value for things that are `null` and we would be making special cases.

I would change this to check for `undefined` if that's what you want and set it to `null`, but not check for `0`.

Reply:
Make sense, zero should be zero